File: AccuPay.Client/src/app/employees/shared/employee-page-options.ts
import { PageOptions } from 'src/app/core/shared/page-options';

export class EmployeePageOptions extends PageOptions {
  constructor(
    pageIndex: number,
    pageSize: number,
    private searchTerm?: string,
    private filter?: string,
    private positionId?: number
  ) {
    super(pageIndex, pageSize);
    this.searchTerm = searchTerm;
    this.filter = filter;
  }

  toObject(): { [param: string]: string } {
    const object = super.toObject();

    if (this.searchTerm != null) {
      object.searchTerm = this.searchTerm;
    }

    if (this.filter != null) {
      object.filter = this.filter;
    }

    if (this.positionId) {
      object.positionId = String(this.positionId);
    }

    return object;
  }
}

File: AccuPay.Core/Helpers/EmployeePageOptions.cs
namespace AccuPay.Core.Helpers
{
    public class EmployeePageOptions : PageOptions
    {
        public string SearchTerm { get; set; }

        public bool HasSearchTerm => !string.IsNullOrWhiteSpace(SearchTerm);

        public string Filter { get; set; }

        public bool HasFilter => !string.IsNullOrWhiteSpace(Filter);

        public int? PositionId { get; set; }

        public bool HasPosition => PositionId.HasValue;
    }
}
